src/lib/pubkey/gost_3410/gost_3410.cpp

```diff
@@ -76,22 +77,18 @@ GOST_3410_PublicKey::GOST_3410_PublicKey(const AlgorithmIdentifier& alg_id, std:
    const size_t part_size = bits.size() / 2;
 
    // Keys are stored in little endian format (WTF)
-   for(size_t i = 0; i != part_size / 2; ++i) {
-      std::swap(bits[i], bits[part_size - 1 - i]);
-      std::swap(bits[part_size + i], bits[2 * part_size - 1 - i]);
-   }
-
-   BigInt x(bits.data(), part_size);
-   BigInt y(&bits[part_size], part_size);
-
-   m_public_key = domain().point(x, y);
+   std::vector<uint8_t> encoding;
+   encoding.reserve(bits.size() + 1);
+   encoding.push_back(0x04);
+   encoding.insert(encoding.end(), bits.rbegin() + part_size, bits.rend());
+   encoding.insert(encoding.end(), bits.rbegin(), bits.rend() - part_size);
```

Comment:
Perhaps that's a good vehicle to try and use `std::ranges::views` (and perhaps even integrate this with `concat()`), to end up with a construction like so:

```C++
namespace v = std::ranges::views;
auto encoding =
   concat<std::vector<uint8_t>>(
      '\x04',
      v::reverse(std::span{bits}.first(part_size)),
      v::reverse(std::span{bits}.last(part_size)));
```

... certainly not here, though.